### PR TITLE
search: add index mappings for text and html content

### PR DESCRIPTION
**Pull request by strib**
And test that it works.

(Based on #21682 -- review that one first.)

Issue: HOTPOT-1488

### CODE REVIEW

File: go/kbfs/search/mapping.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	"github.com/blevesearch/bleve/analysis"
+	"github.com/blevesearch/bleve/analysis/char/html"
+	"github.com/blevesearch/bleve/analysis/token/lowercase"
+	"github.com/blevesearch/bleve/analysis/tokenizer/unicode"
+	"github.com/blevesearch/bleve/mapping"
+	"github.com/blevesearch/bleve/registry"
+)
+
+const (
+	htmlAnalyzerName = "html"
```
Comment (songgao) [inline go/kbfs/search/mapping.go]: Is it a concern bleve will start having a `html` analyzer built-in? It already has a `html` char filter; I wonder why they didn't just make an analyzer. Would `kbfsHtml` be better?

File: go/kbfs/search/mapping.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	htmlFilter, err := cache.CharFilterNamed(html.Name)
+	if err != nil {
+		return nil, err
+	}
+	toLowerFilter, err := cache.TokenFilterNamed(lowercase.Name)
+	if err != nil {
+		return nil, err
+	}
+	rv := analysis.Analyzer{
+		Tokenizer: tokenizer,
```
Comment (songgao) [inline go/kbfs/search/mapping.go]: I wonder if we should add the `web` tokenizer too. Seems useful: https://github.com/blevesearch/bleve/blob/master/analysis/tokenizer/web/web.go

File: go/kbfs/search/mapping.go (hunk 1 above)
Comment (strib) [reply go/kbfs/search/mapping.go]: Sure, fine with me!

File: go/kbfs/search/mapping.go (hunk 2 above)
Comment (strib) [reply go/kbfs/search/mapping.go]: Sure, it looks like it uses unicode under the covers and just adds a few exceptions.  I'll try it.

**Pull request merged**